### DISCUSSION TITLE
Concerns in free project limits

### DISCUSSION THREAD

**Discussion by hyochan** [supabase/supabase#13132] [Questions]
Hello Supabase! I appreciate the excellent alternative you provide to Firebase. I'm a big fan of Firebase, and lately, I've become increasingly drawn to Supabase.

I'd like to express my concerns about the restrictions on the Supabase free tier. It only allows me to create two free projects, which is quite limiting. When developing a product, I typically used Firebase to create two separate projects: one for the staging environment and one for production. This way, I could avoid damaging my database with test data or encountering issues with settings, like social authentication, during development.

With Supabase, however, I'm unable to create a third project for staging. Additionally, I'd like to create sample Supabase apps to showcase its many features and help others improve their productivity. Unfortunately, I had to resort to creating multiple accounts to accomplish this, which doesn't feel ideal.

Is there any chance these limitations could be reconsidered or improved in the future?

**Comment by JSmithOner** [comment]
Why don't you pay or use other email account to separate between your real project ( local (no hosting), staging, production) and another account for supabase examples?

**Reply by hyochan** [reply]
Then I need to sign out and sign in again every time I switch the environment 😢 That is really awkward.

**Comment by GaryAustin1** [comment]
My understanding is Supabase has a non trivial cost for each instance on AWS.  Paying customers basically carry the load for the free tier.
Can you use a local instance as your dev instance?

**Reply by hyochan** [reply]
Yes, I can utilize the local environment. However, it's important to note that a local instance might not perfectly mimic the production environment you'll eventually deploy to.

For example, when using Vercel, there are significant differences between the local and deployed environments, as the latter operates on a serverless architecture.

To ensure reliable deployment, it's advisable to incorporate an additional staging environment. This intermediary step allows you to test and verify the application's behavior under conditions that closely resemble the production environment, minimizing potential issues during deployment.
